src/util.test.js

```diff
@@ -0,0 +1,874 @@
+/*
+* util.js requires Oskari object with few functions in global variables in order to work without errors.
+* For this reason mock object is created and placed to global variables before importing util 
+*/
+const OskariMock = {
+    log: () => {
+        return {
+            // Prevent debug logging in tests to achieve leaner Jest test execution output
+            debug: (text) => { }
+        }
+    },
+    getDecimalSeparator: () => { return '.' }
+};
+
+global.Oskari = OskariMock;
+const util = require('./util');
+const jQuery = require('jquery');
+global.jQuery = jQuery;
+
+describe('throttle function executes given function ', () => {
+
+    test('once immediatelly and once after given wait interval when {leading: false} is not provided', async () => {
+
+        expect.assertions(2);
+
+        const functionCallCount = 5;
+        const wait = 500;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait);
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+            if (i === 0) {
+                expect(mockFunction.mock.calls.length).toEqual(1);
```

Comment:
This should work when moved out of the for loop and called without the if.

Reply:
Mentioned tested is refactored to be more clear by removing unnecessary  for loop.